src/activities/quizzes/timing/QuizTimingEntity.js

```diff
@@ -0,0 +1,76 @@
+import { Entity } from '../../../es6/Entity';
+import { Actions, Classes } from '../../../hypermedia-constants';
+import { performSirenAction } from '../../../es6/SirenAction.js';
+
+/**
+ * A QuizTiming subentity of a d2l Quiz.
+ */
+export class QuizTimingEntity extends Entity {
+	/**
+	 * @returns {bool} Whether or not the edit timing type action is present
+	 */
+	canEditTiming() {
+		return this._entity && this._entity.hasActionByName(Actions.quizzes.timing.updateType);
+	}
+
+	timingTypes() {
+		const action = this._entity && this._entity.getActionByName(Actions.quizzes.timing.updateType);
+		if (!action) return;
+		const field = action.getFieldByName('timingType');
+		if (!field) return;
+		return field.value;
+	}
+
+	submissionLateType() {
+		const entity = this._entity && this._entity.getSubEntityByClass(Classes.quizzes.timing.enforced);
```

Comment:
We could add a helper function to check if it is an `enforced` or `recommended` entity

Reply:
I included two new helper methods: `getEnforcedTimingSubEntity()` and `getRecommendedTimingSubEntity()`. Is that what you meant or did you prefer one helper which checks which subentity is present and to returns that subentity?